### DISCUSSION TITLE
Estimator snaive Not Available

### DISCUSSION THREAD

**Discussion by Spongillas** [pycaret/pycaret#2749] [Q&A] [answered]
Hi!

I get "ValueError: Estimator snaive Not Available. Please see docstring for list of available estimators." when I'm trying to compare models with 'snaive' estimator.

If I replace minimum and maximum values of dataset, estimator works well.
`test_df.loc[test_df['Total']<0.001,'Total']=0.01`
`test_df.loc[test_df['Total']>5,'Total']=5`

Before:
![image](https://user-images.githubusercontent.com/61380413/178843535-c4264451-bcb6-4dd6-b473-54f76ca2e5ae.png)
After:
![image](https://user-images.githubusercontent.com/61380413/178843583-e7181c3f-82aa-4a64-b9ac-8ca38986e08e.png)
`best.get_params()`
`>>{'sp': 24, 'strategy': 'last', 'window_length': None}`

But if I will use sktime NaiveForecaster (as I understand PyCaret use it) algorithm works without any transformation:
![image](https://user-images.githubusercontent.com/61380413/178846047-0839c7b3-5d51-4abe-94ec-2f10cc5a7907.png)

What is the reason of this error?

**Comment by ngupta23** [comment, marked as answer]
I think snaive is disabled when seasonality is not detected in the dataset. Can you check to confirm if seasonality is disabled when setup is run?

**Reply by Spongillas** [reply]
You are right! So simple answer again..
Thanks)

**Comment by shuaiwang88** [comment]
@ngupta23  is there a way to make the snavie work even the seasonality is not detected? 
Or is there a mapping of available algos if seasonality is detected or not? 
I am looping through a list of sales units of say 10 items, where some of them have seasonality and some not.  It gives me error when adding snavie because item x does not have seasonality. 
Maybe adding a condition, say if seasonality is detected, then use one set of algorithms, if not use another set of algorithms. 
I'd like to use fixed 5 algorithms for all the items, so I'd like to pick algo from each category like boosting, normal time series, etc.

**Reply by shuaiwang88** [reply]
solution:
```
        best_models = compare_models(sort='RMSE', verbose=verbose, n_select=3,
                                         include=["snaive", "croston", "rf_cds_dt", "lightgbm_cds_dt"])
        except Exception as e:
            best_models = compare_models(sort='RMSE', verbose=verbose, n_select=3,
                                         include=["croston", "rf_cds_dt", "lightgbm_cds_dt"])
        finally:
            best_tuned_models = [tune_model(model) for model in best_models]
            best_blend_model = blend_models(best_tuned_models, method='mean')
```

**Reply by ngupta23** [reply]
You can check available models using `exp.models()`. If `snaive` is there, use it else use `naive`. This can be done programmatically so you don't have to run compare_models multiple times .
